### ISSUE TITLE
Is it using num_epochs=400？

### ISSUE THREAD

**Issue by kelikeli** [ZihaoW123/UniMM#1]
How long does it take to train with 4 GPUs? Is it using num_epochs=400？

**Comment by ZihaoW123** [comment]
We early stop the training process based on the performance of the model on the val set. It use about 100 epochs for training with sparse annotations. Visdom is used for logging. Specify visdom_server, visdom_port and enable_visdom arguments in options.py to use this feature. 

In each epoch, for an image, we sample only 1 positive instance(_sequences_per_image-num_negative_samples_) and 5 negative samples (_num_negative_samples=5_) from 10 rounds of question-answer pairs. That is, in each epoch, the model learns only 1/10 positive instances of the training set. As a result, it uses a lot of epochs.

**Comment by kelikeli** [comment]
> We early stop the training process based on the performance of the model on the val set. It use about 100 epochs for training with sparse annotations. Visdom is used for logging. Specify visdom_server, visdom_port and enable_visdom arguments in options.py to use this feature.
> 
> In each epoch, for an image, we sample only 1 positive instance(_sequences_per_image-num_negative_samples_) and 5 negative samples (_num_negative_samples=5_) from 10 rounds of question-answer pairs. That is, in each epoch, the model learns only 1/10 positive instances of the training set. As a result, it uses a lot of epochs.

So the following command is correct in the traing?  -num_epochs=100 
python train.py -batch_size 240 -num_epochs 100 -batch_multiply 1 -lr 2e-5 -image_lr 2e-5 -mask_prob 0.15 -sequences_per_image 6 -num_negative_samples 5 -start_path checkpoints-release/vqa_pretrained_weights -visdom_env ugvdce_gen_dis

**Comment by ZihaoW123** [comment]
> 
That's fine, but it doesn't guarantee the best results on the validation set.
